### DISCUSSION TITLE
Failed to get remote IP Address

### DISCUSSION THREAD

**Discussion by Shinefallen** [uNetworking/uWebSockets#1046] [Q&A] [answered]
I tried to get the remote IP address by using ws->getRemoteAddress(), while some garbled codes ( just like ▒▒▒O) was printed. How can i get the IP address correctly? Thanks~

**Comment by ghost** [comment]
It is binary

**Comment by Shinefallen** [comment]
> It is binary

Thanks for your reply. The error message shows the return value of getRemoteAddress() is class std::basic_string_view<char>，so what do you mean it is binary？How can I convert its return value to string. I have tried some methods but failed. I'm new in C++17...

**Comment by ghost** [comment, marked as answer]
IPv4 is 4 bytes IPv6 is 16 bytes. These are the bytes given.

"127.0.0.1" is {127, 0, 0, 1}
